lib/kubernetes-deploy/runner.rb

```diff
@@ -339,32 +338,24 @@ def deploy_resources(resources, prune: false)
     end
 
     def confirm_context_exists
-      out, err, st = run_kubectl("config", "get-contexts", "-o", "name", namespaced: false, with_context: false)
+      out, err, st = kubectl.run("config", "get-contexts", "-o", "name", use_namespace: false, use_context: false)
       available_contexts = out.split("\n")
       if !st.success?
         raise FatalDeploymentError, err
       elsif !available_contexts.include?(@context)
         raise FatalDeploymentError, "Context #{@context} is not available. Valid contexts: #{available_contexts}"
       end
-      KubernetesDeploy.logger.info("Context #{@context} found")
+      @logger.info("Context #{@context} found")
     end
 
     def confirm_namespace_exists
-      _, _, st = run_kubectl("get", "namespace", @namespace, namespaced: false)
+      _, _, st = kubectl.run("get", "namespace", @namespace, use_namespace: false, log_failure: false)
       raise FatalDeploymentError, "Namespace #{@namespace} not found" unless st.success?
-      KubernetesDeploy.logger.info("Namespace #{@namespace} found")
+      @logger.info("Namespace #{@namespace} found")
     end
 
-    def run_kubectl(*args, namespaced: true, with_context: true)
-      if namespaced
-        raise KubectlError, "Namespace missing for namespaced command" if @namespace.blank?
-      end
-
-      if with_context
-        raise KubectlError, "Explicit context is required to run this command" if @context.blank?
-      end
-
-      Kubectl.run_kubectl(*args, namespace: @namespace, context: @context)
+    def kubectl
```

Comment:
OTOH for the Runner, the laziness lets us raise friendlier errors (from `validate_configuration` inside `run` instead of from the initializer building the kubectl instance), and for KubernetesResource, I think it makes more sense for the abstract superclass to provide a ready-to-use method than an ivar that hopefully subclass initializers won't forget to populate.